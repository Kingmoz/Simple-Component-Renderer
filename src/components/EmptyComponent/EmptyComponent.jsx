import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const NotFoundMessage = styled.h1`
  color: red;
`;

const EmptyComponent = ({ componentName, children }) => {
  return (
    <div>
      <NotFoundMessage>
        {
          componentName
            ? `${componentName} not found.`
            : `Component type is missing`
        }
      </NotFoundMessage>
      { children }
    </div>
  );
};

EmptyComponent.propTypes = {
  componentName: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}

export default EmptyComponent;