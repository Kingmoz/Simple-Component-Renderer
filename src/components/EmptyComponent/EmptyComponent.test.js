import React from 'react';
import { mount } from 'enzyme';

import EmptyComponent from './EmptyComponent';

function setup() {
  const props = {
    componentName: 'Box1'
  };

  return mount(<EmptyComponent {...props} />);
}

describe('EmptyComponent', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render componentName correctly', () => {
    expect(wrapper.text()).toEqual('Box1 not found.');
  });

});