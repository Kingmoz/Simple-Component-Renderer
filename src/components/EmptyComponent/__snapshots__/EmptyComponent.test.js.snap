// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyComponent should render properly 1`] = `"<div><h1 class=\\"sc-bdVaJa rOCEJ\\">Box1 not found.</h1></div>"`;
